tests/ReportOutputTest.php

```diff
@@ -1323,4 +1327,96 @@ private function toUnixLineEndings(string $output): string
     {
         return preg_replace('~\r\n?~', "\n", $output);
     }
+
+    /**
+     * @dataProvider outputProvider()
+     */
+    public function testConsoleReportWithPrettyPrint(string $file_contents, $expected_output): void
+    {
+        $this->addFile(
+            'somefile.php',
+            $file_contents
+        );
+
+        $this->analyzeFile('somefile.php', new Context());
+
+        $console_report_options = new ReportOptions();
+        $console_report_options->use_color = false;
+        $console_report_options->show_info = false;
+        $console_report_options->pretty_print_array = true;
+
+        $output = IssueBuffer::getOutput(IssueBuffer::getIssuesData(), $console_report_options);
+        foreach (explode(PHP_EOL, $expected_output) as $line) {
+            $this->assertStringContainsString($line, $output);
+        }
+    }
+
+    /**
+     * @return Generator<string, array<string>>
+     */
+    public function outputProvider(): Generator
+    {
+        $file_contents = <<<'EOT'
+<?php
```

Comment:
Since the master branch requires PHP 7.4, I'd prefer the indented heredoc syntax. Not sure if @orklah has any thoughts on that.

Reply:
I never use heredoc so I don't really know, but yeah, if we can indent that, it's probably for the best

Reply:
Ok!

Reply:
this right ? 
https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc

Reply:
Yep, just indent everything including `EOT`.

Reply:
I'm here!
I'm busy during this period, but I worked on it...

Reply:
Ok, I elaborate first crappy crappy version!
The idea is:

1. Detect from the "issue->message" two arrays, requested and provided 
to perform this operation I create a class "PrettyDetectArray->detect"

This is a complex operation because the message is a "random" payload! 
The next idea might add some sort of meta-data, to the point where detecting the issue. 
In this case, the process is simple: format and compare... (next in work...) 

2. Format one and second array 
to perform this operation I create a class "PrettyFormat->format"

3.  Compare one and second array
to perform this operation I create a class "PrettyCompare->compare"

I think to improve the concrete code, to make it more clean and readable 

push coming

Reply:
> Detect from the "issue->message" two arrays, requested and provided to perform this operation I create a class "PrettyDetectArray->detect"

This seems unnecessarily complicated and will probably cause issues if I do something like `class Myarray` `Myarray{key: 'value'}`. I think it would probably be easier and better to do something like add a `$types` list to issues and have the message be a format string. You could even be more generic and support multiple rendering categories in the future:

```<?php
new SomeIssue('The type "{provided}" doesn't match the expected "{expected}"', ["provided" => $provided_type, "expected" => $expected_type], $suppressed_issues);
```

On the other hand having to track the type in the issue requires more refactoring, so idk. If we do decide to do a refactor requiring a change to issue messages, we might as well try to be consistent about the order of "Foo provided, expected Bar" vs "Expected Foo, found Bar" everywhere.

Really nice to get this improvement though!

Reply:
Thinking about it a bit more, I think if we went the route of having the issues track the actual `Union` types, we'd probably want to make the message generic like `'The provided type doesn't match the expected type'`, and then when it's rendered that can be converted to either `'The provided...type. Provided: array{...}; Expected: array{...}` or the new pretty format based on config and output type.

If you've already gotten the type detection mostly working though, that might be easier to go with for now until things get closer to finished.

Reply:
Nice !
>Really nice to get this improvement though!
I'm ready to go in this direction for me is the best! 

I could create a intermediate task/issue to doing this work, then complete this ...

Reply:
Can you explain the second sentence?
> Thinking about it a bit more, I think if we went the route of having the issues track the actual Union types, ...
like structure with the "tokens" ?